amgcl/detail/qr.hpp

```diff
@@ -223,10 +223,10 @@ class QR {
             const int row_stride = (Order == row_major ? nmax : 1);
             const int col_stride = (Order == row_major ? 1 : m);
 
-            for(int i = 0, ii = 0; i < n; ++i, ii += row_stride + col_stride)
+            for(int i = 0, ii = 0; i < m; ++i, ii += row_stride + col_stride)
```

Comment:
Just to get this right: We should have a single code for QR decomposition, and another for the QR-based solver. The solver should call the QR decomposition directly, but either on `A` for tall matrices or `A^T` for wide matrices. And finally it should solve either `Q.R.x = f` or `R^T.Q^T.x = f`. Correct?